gpt_engineer/data/file_repository.py

```diff
@@ -197,20 +244,93 @@ def __delitem__(self, key: Union[str, Path]) -> None:
         elif item_path.is_dir():
             shutil.rmtree(item_path)
 
+    def _xml_tree_supported_files(self, directory: Path, parent: ET.Element) -> bool:
```

Comment:
I think we can make this much simpler if we 1. first build a nested dict of the directory and 2. then parse it to yaml - see [here](https://gist.github.com/UmerHA/a0845f17325f07c554a6dada9fc0cab2)

We want the codebase to be as easy & accessible as possible, so I'd vote to make this part simpler

Reply:
To be fair i'm not actually using this code at all so i could / should remove it. I'm simply passing in the list of all files which was the other method i created. Like you say it's simpler and easier to work with. The only reason i included xml is that i thought someone might find it useful in the future for maybe getting gpt engineer to move directories etc 

I understand from blogs by sweep ai etc that xml is a good language for LLMs because the large amount of xml data they train on via html 

happy to remove for now though as it's not being used?

Reply:
Have removed it now as it wasn't in use